src/components/Modals/AddToContestModal/AddToContestModal.tsx

```diff
@@ -68,13 +68,17 @@ export default class AddToContestModal extends React.PureComponent<Props, State>
   }
 
   getBaseState(): State {
-    const { contest, currentProject, userEmail } = this.props
+    const { contest, currentProject, userEmail, hasAcceptedTerms } = this.props
     return {
       project: { ...currentProject! },
-      contest: { ...contest, email: contest.email || userEmail || '' }
+      contest: { ...contest, email: contest.email || userEmail || '', hasAcceptedTerms }
     }
   }
 
+  handleToggleTermsAndConditions = (_: any) => {
+    this.setState({ contest: { ...this.state.contest, hasAcceptedTerms: !this.state.contest.hasAcceptedTerms } })
```

Comment:
just to be consistent, maybe we can do:
```suggestion
    const { contest } = this.state
    this.setState({ contest: { ...contest, hasAcceptedTerms: !contest.hasAcceptedTerms } })
```

or
```suggestion
    const { contest } = this.state
    const hasAcceptedTerms = !contest.hasAcceptedTerms
    this.setState({ contest: { ...contest, hasAcceptedTerms } })
```